telluric/georaster.py

```diff
@@ -738,10 +738,21 @@ def __contains__(self, geometry):
         """
         return self.footprint().contains(geometry)
 
-    def astype(self, dst_type, stretch=False):
+    def astype(self, dst_type, in_range='dtype', out_range='dtype', clip_negative=False, stretch=False):
         """ Returns copy of the raster, converted to desired type
-        Supported types: uint8, uint16, uint32, int8, int16, int32
-        For integer types, pixel values are within min/max range of the type.
+        Supported types: uint8, uint16, uint32, int8, int16, int32, float16, float32, float64
+
+        :param dst_type: desired type
+        :param in_range: str or 2-tuple, default 'dtype':
+            'image': use image min/max as the intensity range,
+            'dtype': use min/max of the image's dtype as the intensity range,
+            2-tuple: use explicit min/max intensities
+        :param out_range: str or 2-tuple, default 'dtype':
+            'dtype': use min/max of the image's dtype as the intensity range,
+            2-tuple: use explicit min/max intensities
+        :param clip_negative: boolean, if `True` - clip the negative range, default False
```

Comment:
why do we need this option?

Reply:
If `out_range == 'dtype'` and `clip_negative=True` then clip the negative range even if the output dtype allows negative values. I think that this may be useful in some cases. For example:

```
In [12]: some_array = np.array([[1, 2, 3], [4, 5, 6]], dtype=np.uint8)

In [13]: some_image = np.ma.array(some_array)

In [14]: some_crs = {'init': 'epsg:32620'}

In [15]: some_affine = Affine.translation(3, 2)

In [16]: some_raster = GeoRaster2(some_image, affine=some_affine, crs=some_crs)

In [19]: some_raster.astype(dst_type=np.int8, in_range='image').image
Out[19]: 
masked_array(
  data=[[[-128,  -77,  -26],
         [  25,   76,  127]]],
  mask=False,
  fill_value=999999,
  dtype=int8)

In [20]: some_raster.astype(dst_type=np.int8, in_range='image', clip_negative=True).image
Out[20]: 
masked_array(
  data=[[[  0,  25,  50],
         [ 76, 101, 127]]],
  mask=False,
  fill_value=999999,
  dtype=int8)
```